engine-src/Game/LambdaHack/Client/UI/FrameM.hs

```diff
@@ -168,43 +173,37 @@ promptGetKey dm ovs onBlank frontKeyKeys = do
   modifySession $ \sess ->
     sess { sdisplayNeeded = False
          , sturnDisplayed = True
-         , smacroStack = addToMacro brevMap km $ smacroStack sess }
+         , sactionPending = addToMacro brevMap km $ sactionPending sess }
   return km
 
-addToMacro :: M.Map HumanCmd.HumanCmd [K.KM] -> K.KM -> MacroStack -> MacroStack
-addToMacro brevMap km macros =
+addToMacro :: M.Map HumanCmd.HumanCmd [K.KM] -> K.KM -> [ActionBuffer]
+           -> [ActionBuffer]
+addToMacro _ _ [] = error "cannot add macro to empty action stack"
+addToMacro brevMap km (abuff : abuffs) =
   let mKeyRecord = case M.lookup HumanCmd.Record brevMap of
         Nothing -> Nothing
         Just (k : _) -> Just k
```

Comment:
Actually, here I has sloppy, we probably need to take the whole list and check if the key is on that list, instead of considering only the first element.